src/riak_ql_lexer.xrl

```diff
@@ -177,19 +176,31 @@ lex(String) ->
     {ok, Toks, _} = string(String),
     Toks.
 
+clean_up_identifier(Literal) ->
+    clean_up_literal(Literal).
+
 clean_up_literal(Literal) ->
-    RemovedOutsideQuotes = accurate_strip(Literal, $'),
-    DeDoubledInternalQuotes = re:replace(RemovedOutsideQuotes,
-                                         "''", "'",
-                                         [global, {return, list}]),
-    list_to_binary(DeDoubledInternalQuotes).
-
-strip_quoted(QuotedString) ->
-    % if there are unicode characters in the string, throw an error
-    [error(unicode_in_quotes) || U <- QuotedString, U > 127],
-
-    StrippedOutsideQuotes = accurate_strip(QuotedString, $"),
-    re:replace(StrippedOutsideQuotes, "\"\"", "\"", [global, {return, binary}]).
+    Literal1 = case hd(Literal) of
+        $' -> accurate_strip(Literal, $');
```

Comment:
Because `CHARACTER_LITERAL` is now obligated to be single-quoted, you can simply and unconditionally drop the first and last characters of `Literal` here (and remove `accurate_strip/1`).